### PR TITLE
Bundle configuration based on omniplay classes names

### PR DESCRIPTION
**Pull request by jsampedro77**
Modifies bundle configuration:
1. Configuration root is sylius_omnipay, replacing old sylius_payments
2. Available Gateways are retrieved using Omipay GatewayFactory::find
3. In configuration a valid gateway type must be provided, it has to match Omnipay name. Example

```
sylius_omnipay:
      gateways:
        paypal:
          type: PayPal_Express
          label: Paypal Express
        sagepay:
          type: SagePay_Direct
          label: Sagepay Direct
```

I am starting to work with this bundle, hope this modification won't break anything!

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Hey! Looks like great change (and fix). Thanks a lot! Any reason to change the indentation inside the Configuration class, or it's accident? :)

**Comment by jsampedro77** [comment]
ooops, it's netbeans auto-formatting, I didn't realize.

Will restore it soon!

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -21,118 +23,69 @@
...
      * {@inheritdoc}
      */
     public function getConfigTreeBuilder()
     {
-        $builder  = new TreeBuilder();
-        $rootNode = $builder->root('sylius_payments');
+        $builder = new TreeBuilder();
+        $rootNode = $builder->root('sylius_omnipay');
+
+        $gateways = $this->getGateways();
```
Comment (stloyd) [inline DependencyInjection/Configuration.php]: There is no point for this. Variable is accessible internally in class, so also that function can be removed.

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -21,118 +23,69 @@
...
-        'pin',
-        'stripe',
-        'twocheckout',
-        'worldpay',
-    );
+    private $gateways;
+
+    public function __construct()
+    {
+        $this->gateways = OmnipayGatewayFactory::find();
```
Comment (stloyd) [inline DependencyInjection/Configuration.php]: Maybe move this to builder function as it's not doing anything more.

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -13,6 +13,8 @@
 
 use Symfony\Component\Config\Definition\Builder\TreeBuilder;
 use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Omnipay\Common\GatewayFactory as OmnipayGatewayFactory;
+use Omnipay\Common\Helper as OmnipayHelper;
```
Comment (stloyd) [inline DependencyInjection/Configuration.php]: This is never used.

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -21,118 +23,69 @@
  */
 class Configuration implements ConfigurationInterface
 {
+
```
Comment (stloyd) [inline DependencyInjection/Configuration.php]: Please revert those extra new lines in all files.

File: DependencyInjection/SyliusOmnipayExtension.php [hunk 5]
```diff
@@ -17,6 +17,7 @@
 use Symfony\Component\DependencyInjection\Definition;
 use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
 use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+use Omnipay\Common\Helper as OmnipayHelper;
```
Comment (stloyd) [inline DependencyInjection/SyliusOmnipayExtension.php]: What's point of using alias ? It's not conflicting with any used class here.

**Comment by pjedrzejewski** [comment]
Thanks @jsampedro77! And @stloyd for review. Merged! I'll tweak some minor issues later.

**Pull request merged**